packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts

```diff
@@ -226,18 +230,9 @@ class FirebaseAnalyticsJS {
     userPropertyName: string,
     userPropertyValue: any
   ): string {
-    if (
-      !userPropertyName.length ||
-      userPropertyName.length > 24 ||
-      userPropertyName[0] === '_' ||
-      !userPropertyName.match(/^[A-Za-z_]+$/) ||
-      userPropertyName === 'user_id' ||
-      userPropertyName.startsWith('firebase_') ||
-      userPropertyName.startsWith('google_') ||
-      userPropertyName.startsWith('ga_')
-    ) {
+    if (!FirebaseAnalyticsJS.isValidName(userPropertyName, 24)) {
```

Comment:
Perhaps we should add the check for the reserved `user_id` name explicitly here.

```suggestion
    if (!FirebaseAnalyticsJS.isValidName(userPropertyName, 24) || (userPropertyName === 'user_id')) {
```